astropy/io/fits/tests/test_core.py

```diff
@@ -1241,8 +1241,9 @@ def get_free_space_in_dir(path):
                 hdulist.writeto(fileobj)
 
         assert (
-            "Not enough space on disk: requested 8000, available 0. "
-            "Fake error raised when writing file." == exc.value.args[0]
+            exc.value.args[0]
+            == "Not enough space on disk: requested 8000, available 0. "
+            "Fake error raised when writing file."
```

Comment:
I would rather use `pytest.raises(..., match=...)` than check the error message in a separate `assert`. I'm only pointing it out here, but there are a couple more cases like this.

Reply:
I'm happy to refactor this code.

Reply:
As in the ones affected by this PR. For other cases, that's out of scope of this PR :)

Reply:
Note that ruff error code `PT011` will catch all of the missing `match` argument.

Reply:
I refactored the ones touched by this PR, so the check will be happier by a few.